problems/problems_controller.go

```diff
@@ -27,9 +31,37 @@ func (t *Controller) SetIndex(w http.ResponseWriter, r *http.Request, _ httprout
 	w.Write(data)
 }
 
+// GetUserProblem handles the /users/:userId/problems/:problemId endpoint, returns problem details
+// for a given user.
+func (t *Controller) GetProblem(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
+	w.Header().Set("Content-Type", "application/json")
+
+	userID := params.ByName("userId")
+	user, err := t.usersDataProvider.FindUser(userID)
+	if err != nil {
+		httpresp.Error(w, "User not found", http.StatusNotFound)
+		return
+	}
+
+	problemID, err  := strconv.Atoi(params.ByName("problemId"))
+	if err != nil {
+		httpresp.BadRequest(w, "Invalid problem ID")
+	}
+
+	problem, err := t.problemsDataProvider.FindProblem(user.UserName, uint(problemID))
+	if err != nil {
+		log.Printf("Failed to retrieve problem: %s", err)
```

Comment:
Should we include `UserName` and `problemID` on the log?

Reply:
Sounds like a good idea.